### PR TITLE
Feature/17902 check forms expand sorting

### PR DESCRIPTION
**Pull request by pris54**

### CODE REVIEW

File: admin/spec/controllers/check-form.spec.js [hunk 1]
```diff
@@ -196,15 +304,70 @@ describe('check-form controller:', () => {
...
+        }
+
+        expect(next).not.toHaveBeenCalled()
         expect(res.locals.pageTitle).toBe('View form')
+        expect(res.statusCode).toBe(302)
+        done()
+      })
+
+      it('should catch an error when getCheckWindowsAssignedToForms fails', async (done) => {
+        spyOn(checkFormDataService, 'getActiveFormPlain').and.returnValue(function () { return Promise.resolve(checkFormMock) })
```
Comment (jon-shipley) [inline admin/spec/controllers/check-form.spec.js]: Is this meant to be returning a function?

File: admin/services/data-access/check-window.data.service.js [hunk 2]
```diff
@@ -37,9 +37,10 @@ const checkWindowDataService = {
     query.isDeleted = !deleted ? false : deleted
     if (current === true) {
       query.checkEndDate = {$gte: currentTimestamp}
-    } else {
-      query.checkEndDate = {$lt: currentTimestamp}
     }
+    // else {
```
Comment (GeorgeChatzigiannis) [inline admin/services/data-access/check-window.data.service.js]: do you need those commented lines?

**Comment by jon-shipley** [comment]
`standard` is reporting some lint fluff on 
* `check-form.spec.js`
* `check-window.spec.js`
* `check-form.service.spec.js`
* `check-window.service.spec.js`

File: admin/spec/controllers/check-form.spec.js [hunk 3]
```diff
@@ -119,6 +142,34 @@ describe('check-form controller:', () => {
         expect(checkFormService.unassignedCheckFormsFromCheckWindows).toHaveBeenCalled()
         done()
       })
+
+      it('should fail when getActiveForm method fails', async (done) => {
+        spyOn(checkFormDataService, 'getActiveForm').and.returnValue(Promise.reject(new Error('error')))
+        spyOn(checkWindowService, 'getCheckWindowsAssignedToForms').and.returnValue()
+
+        await controller(req, res, next)
+        expect(res.statusCode).toBe(200)
```
Comment (GeorgeChatzigiannis) [inline admin/spec/controllers/check-form.spec.js]: would that be the right status code in that case?

File: admin/spec/controllers/check-form.spec.js [hunk 4]
```diff
@@ -196,15 +304,70 @@ describe('check-form controller:', () => {
...
+        spyOn(checkFormService, 'checkWindowNames').and.returnValue('Check Window 1')
+        spyOn(checkFormService, 'canDelete').and.returnValue(false)
+        spyOn(console, 'log').and.returnValue('test')
+
+        controller = proxyquire('../../controllers/check-form', {}).displayCheckForm
+      })
+
+      it('should catch an error when getActiveFormPlain fails', async (done) => {
+        spyOn(checkFormDataService, 'getActiveFormPlain').and.returnValue(Promise.reject(new Error('error')))
+        spyOn(checkWindowService, 'getCheckWindowsAssignedToForms').and.returnValue(function () { return Promise.resolve(checkFormsByWindowMock) })
```
Comment (GeorgeChatzigiannis) [inline admin/spec/controllers/check-form.spec.js]: Same question as Jon made below

File: admin/spec/service/check-window.service.spec.js [hunk 5]
```diff
@@ -124,25 +124,18 @@ describe('check-window.service', () => {
...
+      it('should return an error', async (done) => {
         try {
           const result = await service.markAsDeleted(checkFormMock)
           expect(result).toBeTruthy()
-          done()
         } catch (error) {
-          console.log('ERROR!', error)
-          expect(error.toString()).toBe('ERROR SAVING')
-          done()
+          expect(error.toString()).toBe('Error: This form does not have an id')
```
[the remark this answers is not in the document]
Comment (pris54) [reply admin/spec/service/check-window.service.spec.js]: No, we don't *have* to.

File: admin/services/data-access/check-window.data.service.js [hunk 6]
```diff
@@ -32,14 +32,15 @@ const checkWindowDataService = {
     let sorting = {}
     let query = {}
 
-    const currentTimestamp = moment.utc(Date.now()).format('YYYY-MM-D HH:mm:ss.SSS')
+    const currentTimestamp = moment.utc(Date.now()).format('YYYY-MM-D 00:00:00')
```
Comment (jon-shipley) [inline admin/services/data-access/check-window.data.service.js]: Could we not use the DateService here?

**Pull request merged**